File: src/modules/algorithm/seed/algorithms/dijkstra-list/typescript/exercise.ts
/**
 * Interface for a weighted graph edge
 */
interface Edge {
  node: string;
  weight: number;
}

/**
 * Type definition for an adjacency list representation of a graph
 * where keys are node names and values are arrays of edges
 */
type Graph = {
  [key: string]: Edge[];
};

/**
 * Type definition for the result of Dijkstra's algorithm
 * where keys are node names and values are the shortest distance from the source
 */
type DistanceMap = {
  [key: string]: number;
};

/**
 * Implements Dijkstra's algorithm to find the shortest path from a source node
 * to all other nodes in a weighted graph.
 *
 * @param graph - An adjacency list representation of a weighted graph
 * @param source - The source node to start the algorithm from
 * @returns A map of nodes to their shortest distance from the source
 */
export function dijkstra(graph: Graph, source: string): DistanceMap {
  // TODO: Implement Dijkstra's algorithm
}

File: src/modules/algorithm/seed/algorithms/dijkstra-list/go/exercise.go
package dijkstra

// Edge represents a weighted edge in the graph
type Edge struct {
	Node   string
	Weight int
}

// Graph is an adjacency list representation of a weighted graph
type Graph map[string][]Edge

// Dijkstra implements Dijkstra's algorithm to find the shortest path from a source node
// to all other nodes in a weighted graph.
//
// Parameters:
//   - graph: An adjacency list representation of a weighted graph
//   - source: The source node to start the algorithm from
//
// Returns:
//   - A map of nodes to their shortest distance from the source
func Dijkstra(graph Graph, source string) map[string]float64 {
	// TODO: Implement Dijkstra's algorithm
	return nil
} 